src/ui/public/doc_table/components/table_row.js

```diff
@@ -158,6 +163,22 @@ module.directive('kbnTableRow', function ($compile) {
 
         return text;
       }
+
+      function truncateSourceSummary(sourceElement) {
+        const sourceList = sourceElement.find('dl');
+        const listItems = sourceList.children();
+        const listPairs = _.chunk(listItems, 2);
+        sourceList.empty();
+
+        let length = 0;
+        _.forEach(listPairs, (listPair) => {
+          length += (listPair[0].textContent.length + listPair[1].textContent.length);
+          if (length < 500) {
+            sourceList.append(listPair);
+            sourceList.append(' ');
+          }
+        });
```

Comment:
Yes, you're correct. :blush: Still, since you're optimizing for performance just pushing individual items onto the collection might be costly. Probably depends on how this is implemented internally in jQuery/jqLite.